b2/account_info.py

```diff
@@ -141,6 +141,42 @@ def clear_large_file_upload_urls(self, file_id):
         pass
 
 
+class UploadUrlPool(object):
+    """
+    For each key (either a bucket id or large file id), holds a pool
+    of (url, auth_token) pairs, with thread-safe methods to add and
+    remove them.
+    """
+
+    def __init__(self):
+        self._lock = threading.Lock()
+        self._pool = collections.defaultdict(list)
+
+    def put(self, key, url, auth_token):
+        """
+        Adds the url and auth token to the pool for the given key.
+        """
+        with self._lock:
+            pair = (url, auth_token)
+            self._pool[key].append(pair)
+
+    def take(self, key):
+        """
+        Returns (url, auth_token) if one is available, or (None, None) if not.
+        """
+        with self._lock:
+            pair_list = self._pool[key]
+            if len(pair_list) == 0:
```

Comment:
I do. When I read this (python) code, I am wondering why the author is explicitly checking if the length of the pair_list is zero, instead of doing it as it is always done in python.